examples/repo_example/succinct_hash_bin_delegations.py

```diff
@@ -0,0 +1,226 @@
+# Copyright New York University and the TUF contributors
+# SPDX-License-Identifier: MIT OR Apache-2.0
+"""
+A TUF succinct hash bin delegation example using the low-level TUF Metadata API.
+
+The example code in this file demonstrates how to *manually* perform succinct
+hash bin delegation using the low-level Metadata API.
+Succinct hash bin delegation achieves a similar result as using a standard hash
+bin delegation, but the delegated json files are smaller resulting in fewer
+bytes to transfer and parse.
+
+See 'basic_repo.py' for a more comprehensive TUF metadata API example.
+
+For a comprehensive explanation of succinct hash bin delegation and the
+difference between succinct and standard hash bin delegation read:
+https://github.com/theupdateframework/taps/blob/master/tap15.md
+
+NOTE: Metadata files will be written to a 'tmp*'-directory in CWD.
+"""
+import math
+import os
+import tempfile
+from datetime import datetime, timedelta
+from pathlib import Path
+from typing import Dict, List, Tuple
+
+from securesystemslib.keys import generate_ed25519_key
+from securesystemslib.signer import SSlibSigner
+
+from tuf.api.metadata import (
+    Delegations,
+    Key,
+    Metadata,
+    SuccinctRoles,
+    TargetFile,
+    Targets,
+)
+from tuf.api.serialization.json import JSONSerializer
+
+# Succinct hash bin delegation
+# ===================
+# As mentioned above, succinct hash bin delegation has the same purpose as
+# standard hash bin delegation - distribute a large number of target files over
+# multiple delegated targets metadata roles. The consequence is a smaller
+# metadata files and thus a lower network overhead for repository-client
+# communication.
+#
+# The assignment of target files to a target's metadata is done automatically,
+# based on the byte digest of the target file name. More precisely, only a
+# prefix of the target file name byte digest is needed to assign it to the
+# correct hash bin.
+#
+# In succinct hash bin delegation all delegated roles (bins) are represented by
+# one single "succinct_roles" dictionary or when a targets file is loaded with a
+# SuccinctRoles instance from tuf/api/metadata.py
```

Comment:
- I think *"all delegated roles (bins) are represented by one single "succinct_roles" dictionary"* could be misinterpreted. E.g. do you mean that each bin metadata is in this dictionary?
- I don't exactly know what you mean by: *"or when a targets file is loaded with a SuccinctRoles instance from tuf/api/metadata.py"*
- I don't think you need to mention ` tuf/api/metadata.py` when you talk about `SuccinctRoles`, because the class is imported into local scope. (same comment applies elsewhere).

Can we omit the whole paragraph here? The first instance of `SuccinctRoles` we create is another 100 lines away. Maybe we can talk about it only then.

Reply:
Yes, will remove the paragraph as it's far from an actual `SuccinctRoles` usage.